Add panels support (#641)
pydata/pydata-sphinx-theme: docs/demo/kitchen-sink/web-components.rst

.. INSPIRED FROM sphinx-design documentation

Web components
==============

> Cards and tabs are a key functionalities in many documentations depending on pydata-sphinx-theme. Both `sphinx-design <https://sphinx-design.readthedocs.io/en/latest/index.html>`__ and `sphinx-panels <https://sphinx-panels.readthedocs.io/en/latest/>`__ can be used with this theme. We overwrote some of their display features to make them compatible with our supported themes. For usage, please refer to their documentations.

Reviewer: ```suggestion
Cards and tabs provide some extra UI flexibility for your content. Both `sphinx-design <https://sphinx-design.readthedocs.io/en/latest/index.html>`__ and `sphinx-panels <https://sphinx-panels.readthedocs.io/en/latest/>`__ can be used with this theme. This theme provides custom CSS to ensure that their look and feel is consistent with this theme.

.. seealso::

   For more about how to use these extensions, see `the sphinx-design documentation <https://sphinx-design.readthedocs.io/en/latest/index.html>`_.
   
.. warning::
   
   ``sphinx-panels`` is no longer maintained and recommend you switch to ``sphinx-design``.
   We will deprecate support for sphinx-panels soon.
```

.. danger::

>     To use the :code:`sphinx-panels` extention, you need to integrate theses lines to overwrite the shadows of the panels in your custom :code:`.css` file:

Reviewer: ```suggestion
    To use the :code:`sphinx-panels` extention, add these lines to your custom CSS to overwrite the shadows of the panels:
```

    .. code-block:: css

        /* overwrite panels shadows using pydata-sphinx-theme variable */
        .shadow {
            box-shadow: 0 0.5rem 1rem var(--pst-color-shadow) !important;
        }

    This modification is not needed when using the :code:`sphinx-design` extention.

Here you can find some examples of the **cards** and **tabs** created with the :code:`sphinx-design` extention.

Cards
-----

.. grid::

    .. grid-item-card:: Content of the first card

    .. grid-item-card:: Content of the second card

        :bdg-primary:`example`

    .. grid-item-card::

>         .. dropdown:: :fa:`eye,mr-1` third card

Reviewer: There seems to be an awkward "double outline" effect on this card, can we figure out how to remove that?
